### PR TITLE
Poll status for downloads

### PR DESCRIPTION
**Pull request by mcauer**

### CODE REVIEW

File: sketch_map_tool/routes.py [hunk 1]
```diff
@@ -85,9 +97,22 @@ def status(uuid: str, type_: Literal["quality-report", "sketch-map"]) -> Dict[st
...
+    body = {"id": uuid, "status": task.status, "type": type_}
+    if task.status == SUCCESS:
+        http_status = 200
+        body["href"] = "/api/download/" + uuid + "/" + type_
+    elif task.status in [PENDING, RETRY, RECEIVED, STARTED]:
+        http_status = 202
+    elif task.status in [REJECTED, REVOKED, None, FAILURE]:
+        http_status = 500
+    else:
+        http_status = 500
```
Comment (matthiasschaub) [inline sketch_map_tool/routes.py:113]:
```suggestion
    else:  # Incl. REJECTED, REVOKED, FAILURE
        http_status = 500
```

File: sketch_map_tool/routes.py [hunk 2]
```diff
@@ -85,9 +97,22 @@ def status(uuid: str, type_: Literal["quality-report", "sketch-map"]) -> Dict[st
...
 
-    return {"id": uuid, "status": task.status}
+    return body, http_status
```
Comment (matthiasschaub) [inline sketch_map_tool/routes.py:115]:
```suggestion
    Response(body, status=http_status, mimetype='application/json')
```

Comment (matthiasschaub) [reply sketch_map_tool/routes.py:115]: I find the explicit version better to read. Its somehow assuring to return the Response Object instead just the tuple...

**Review by matthiasschaub — DISMISSED** [review]
Looks good! I just left some minor comments regarding the Python code. But it could be merged as it is.

I like the throughout documentation of the JS functions. That makes it much easier for me to understand/review those. Thanks.

File: sketch_map_tool/routes.py [hunk 3]
```diff
@@ -85,9 +97,22 @@ def status(uuid: str, type_: Literal["quality-report", "sketch-map"]) -> Dict[st
...
+
+    # see celery states and their precedence here:
+    # https://docs.celeryq.dev/en/stable/_modules/celery/states.html#precedence
+    body = {"id": uuid, "status": task.status, "type": type_}
+    if task.status == SUCCESS:
+        http_status = 200
+        body["href"] = "/api/download/" + uuid + "/" + type_
+    elif task.status in [PENDING, RETRY, RECEIVED, STARTED]:
+        http_status = 202
+    elif task.status in [REJECTED, REVOKED, None, FAILURE]:
```
[the remark this answers is not in the document]
Comment (mcauer) [reply sketch_map_tool/routes.py]: Only because it was mentioned in the sources of celery, but it is now anyway covered in the else branch

**Pull request merged**